tests/gateway.rs

```diff
@@ -0,0 +1,62 @@
+use k8s_gateway_api::v1alpha2::{
```

Comment:
We may want to feature-gate these tests:
```suggestion
#![cfg(feature = "cluster")]

use k8s_gateway_api::v1alpha2::{
```

Or better yet, create a separate `integration` crate in this repo so these tests don't always run when checking the types.